spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -78,22 +75,16 @@ private void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consum
 		}
 
 		try {
-			Message message;
-			if (this.messageConverter != null) {
-				message = this.messageConverter.toMessage(pubsubMessage.getData().toByteArray(),
-						new MessageHeaders(messageHeaders));
-			}
-			else {
-				message = MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())
-						.copyHeaders(messageHeaders).build();
-			}
-			sendMessage(message);
+			sendMessage(MessageBuilder.withPayload(
+					this.pubSubTemplate.getMessageConverter().fromPubSubMessage(pubsubMessage, this.payloadType))
+					.copyHeaders(messageHeaders)
+					.build());
 		}
-		catch (RuntimeException re) {
+		catch (Exception re) {
 			if (this.ackMode == AckMode.AUTO) {
 				consumer.nack();
 			}
-			throw re;
+			throw new RuntimeException(re);
```

Comment:
May we throw `PubSubException` here to make it easier to trace logs for errors?

Reply:
Typically this would fail due to message conversion, not necessarily an issue with Pub/Sub per se, but we can do that. I also wounder if we should just make the `PubSubMessageConversionException` inherit from `PubSubException`.

Reply:
Well, that's not true. 
You see, you wrap here even `sendMessage()` and this involves the whole Spring Integration flow downstream. So, anything can happen there: starting with any `MessagingException` and ending with the `TransactionTimedOutException` and so on.

Therefore I would insist to wrap it into the generic `PubSubException` and proceed. Just because there downstream can be anything and not only conversion, but with this specific `PubSubException` we are going to make logs more traceable.

Am I missing anything?

Reply:
BTW, if you really want a fine-grained tracing, you really can make that `PubSubMessageConversionException` as a `PubSubException` and extract conversion to its own `try...catch()`, but again: the `sendMessage()` is a different story and has to be treated respectively.

Reply:
I see. I'll just wrap into `PubSubException` for now like you suggested.